### PR TITLE
Feature/bma/arch fixes

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

Fix some issues in our inject class members. There are some other issues, but I stop here for now.

## Motivation and context

Avoid duplicated code, avoid injecting Session in the Views (Activity/Fragment)

## Screenshots / GIFs

NA

## Tests

- Smoke tests.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/permalink/PermalinkUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+package im.vector.app.features.permalink
+
+import org.matrix.android.sdk.api.session.Session
+import javax.inject.Inject
+
+/**
+ * Contains synchronous methods to create permalinks from the Session.
+ */
+class PermalinkUseCase @Inject constructor(
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/permalink/PermalinkUseCase.kt:25]: Thats a bit weird, a UseCase should only offer one method, as it's supposed to encapsulate one business rule

File: vector/src/main/java/im/vector/app/core/platform/VectorBaseActivity.kt [hunk 2]
```diff
@@ -149,26 +149,22 @@ abstract class VectorBaseActivity<VB : ViewBinding> : AppCompatActivity(), Maver
      * ========================================================================================== */
 
     private lateinit var configurationViewModel: ConfigurationViewModel
-    private lateinit var sessionListener: SessionListener
-    protected lateinit var bugReporter: BugReporter
-    private lateinit var pinLocker: PinLocker
 
+    @Inject lateinit var sessionListener: SessionListener
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/core/platform/VectorBaseActivity.kt:154]: Some of the properties were kept private on purpose, we are losing this. I don't like having @Inject in Base Activity/Fragment class

File: vector/src/main/java/im/vector/app/features/permalink/PermalinkUseCase.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/permalink/PermalinkUseCase.kt:25]: Yes, this is not really a UseCase, rather a Proxy. Which name do you suggest?

File: vector/src/main/java/im/vector/app/core/platform/VectorBaseActivity.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/core/platform/VectorBaseActivity.kt:154]: with @Inject in Views, nothing can be private... I will update for the `SessionListener` instance.

**Pull request merged**